atlasclient/models.py

```diff
@@ -846,3 +846,62 @@ class SearchFulltext(base.QueryableModel):
     relationships = {'entities': Entity,
                      'attributes': AttributeDef,
                      'fullTextResults': FullTextResult}
+
+
+class SearchSavedCollection(base.QueryableModelCollection):
+    """We can also remove this part but just keeping same structure as above"""
+    def load(self, response):
+        for item in response:
+            model = self.model_class(self,
+                                     href=item.get('href'))
+            model.load(item)
+            self._models.append(model)
+
+
+class SearchSaved(base.QueryableModel):
+    collection_class = SearchSavedCollection
+    path = 'search/saved'
+    data_key = 'search_saved'
+    primary_key = 'guid'
+    fields = ('name', 'searchParameters', 'ownerName', 'searchType', 'uiParameters')
+
+    def inflate(self):
+        """Load the resource from the server, if not already loaded. Removed params check for searchParameters"""
+        if not self._is_inflated:
+            if self._is_inflating:
+                #  catch infinite recursion when attempting to inflate
+                #  an object that doesn't have enough data to inflate
+                msg = ("There is not enough data to inflate this object.  "
+                       "Need either an href: {} or a {}: {}")
+                msg = msg.format(self._href, self.primary_key, self._data.get(self.primary_key))
+                raise exceptions.ClientError(msg)
+
+            self._is_inflating = True
+
+            try:
+                params = {}
+                # To keep the method same as the original request. The default is GET
+                self.load(self.client.request(self.method, self.url, **params))
+            except Exception:
+                self.load(self._data)
+
+            self._is_inflated = True
+            self._is_inflating = False
+        return self
+
+    @events.evented
+    def create(self, data, **kwargs):
```

Comment:
same for this, any particular reason to override this ?

Reply:
This model doesn't have any entity type in the response/payload (like there is in entity type) so without overriding this method or _generate_input_dict I can't pass dictionary data to QueryableModel.create method.

As mentioned below are two probable solution to this -

1. Override _generate_input_dict to get the actual json payload
`def _generate_input_dict(self, **kwargs):
        return kwargs['data']`
2. Override create method in SearchSaved

Please let me know your though on this.

Reply:
but what do you override? This is same as the parent `create()`

Reply:
This is not same as parent `create(self, *arg, **kwargs)`. To create saved search my idea was to use dictionary /json payload. And the only way to pass that raw data to REST api is overriding the method (option 2) - `create(self, data)`, otherwise parent create method will pass `{'data' : payload}`.

Please let me know your idea to achieve the same and I will change accordingly.

Here is the document I created - 

> To create saved search for the Atlas user by name::

    payload = """{
    "name": "trying",
    "ownerName": "svc_data_catalog_api",
    "searchType": "BASIC",
    "searchParameters": {
        "typeName": "rdbms_db",
        "excludeDeletedEntities": true,
        "includeClassificationAttributes": false,
        "includeSubTypes": true,
        "includeSubClassifications": true,
        "limit": 0,
        "offset": 0
    },
        "uiParameters": "Select::0,Name::1,Owner::2,Description::3,Type::4,Classifications::5,Term::6,Db::7"
    }"""

    response = client.search_saved.create(data=json.loads(payload))

>